exchanges/graphcache/src/cacheExchange.ts

```diff
@@ -73,21 +73,24 @@ const toRequestPolicy = (
   },
 });
 
-export interface CacheExchangeOpts {
-  updates?: Partial<UpdatesConfig>;
-  resolvers?: ResolverConfig;
-  optimistic?: OptimisticMutationConfig;
-  keys?: KeyingConfig;
+export interface CacheExchangeOpts<Updaters, Resolvers, Optimistic, Keys> {
```

Comment:
This should keep that type safety in there

```suggestion
export interface CacheExchangeOpts<Updaters extends UpdatesConfig, Resolvers extends ResolverConfig, Optimistic extends OptimisticMutationConfig, Keys extends KeyingConfig> {
```